cmd/run/run.go

```diff
@@ -422,6 +425,13 @@ func (s *ServerContext) Run(ctx context.Context, config *serverconfig.Config) er
 		),
 	)
 
+	if config.HTTP.UpstreamTimeout > 0 {
+		timeoutMiddleware := middleware.NewTimeoutHandler(config.HTTP.UpstreamTimeout, s.Logger)
+
+		serverOpts = append(serverOpts, grpc.ChainUnaryInterceptor(timeoutMiddleware.NewUnaryTimeoutInterceptor()))
```

Comment:
We shouldn't be appending gRPC timeout middleware if `config.HTTP.UpstreamTimeout` is configured. That config is specific to the HTTP reverse proxy. We need timeout semantics for gRPC and HTTP, and the config shouldn't be so specific to HTTP.

Please adjust this such that we have a top-level `config.RequestTimeout`  and below where `if config.HTTP.Enabled` then we set `runtime.DefaultContextTimeout` to `config.RequestTimeout + 3s`